coding-concepts.js

```diff
@@ -7,48 +7,48 @@
 
 // --------------------1) What will this log?
 
 const cohort = "LEARN 2023"
-// console.log(cohort.length)
+console.log(cohort.length)
 
-// a) Your answer:
+// a) Your answer: This will show us the length of the string "LEARN 2023".
 // b) Verify and explain:
-
+// I logged the method and it came back as 10. 10 is the amount of characters in the string including the quotations on either side.
 // --------------------2) What will this log?
 
 const greeting = "Hello World!"
-// console.log(greeting[4])
+console.log(greeting[4])
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: This will return "o"
+// b) Verify and explain: We get the "l" in the string "Hello World" because when logged the 4 in the parameter [] tells the log to find the fourth index and since the index starts at zero which would be the "H" and the fourth would be the first "o" that it would be the output.
 
 // --------------------3) What will this log?
 
 const languages = ["JavaScript", "Ruby", "Python", "C++"]
 const index = 1
-// console.log(languages[index])
+ console.log(languages[index])
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: This will show us "Ruby"
+// b) Verify and explain:The reason we get ruby is this, we defined a const which declares the variable index = "Whatever value here" when logged whatever value is in the index will determine what value is returned. We place 1 there and so it counts from index 0 to index 1 to find that value or in this case string "Ruby".  
 
 // --------------------4) What will this log?
 
 const weekendDays = ["saturday", "sunday"]
-// console.log(weekendDays.toUpperCase())
+//console.log(weekendDays.toUpperCase()) //TypeError: weekendDays.toUpperCase is not a function
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: This will output an error 
+// b) Verify and explain: this resulted in a error because we can't use this code as it is. We can't use the .toUpperCase on an array instead we would need to use a for loop to solve this problem. I did have to research this problem because I couldn't understand this on my own.  
 
 // --------------------5) What will this log?
 
 const dataTypes = ["number", "string", "Boolean", "undefined"]
-// console.log(typeof dataTypes.length)
+console.log(typeof dataTypes.length)
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: This will log the string "number"
+// b) Verify and explain: The reason why this show number is that the typeof operator will always show a string and when used in tandem with the .length method will show the first string in the index which is "number".  I'm gonna be honest, I had to research this one as well cause I didn't fully understand why the typeof does that.
```

Comment:
The `typeof` operator will return what the data type of the value right after it is. So we have `dataTypes.length`, which will return the length of the array, which is `4`. Then, `typeof` looks at that `4`, and returns its data type, which is a number.